### PR TITLE
UIDATIMP-825: Refactor File Extensions view to use final-form instead of redux-form

### PR DESCRIPTION
**Pull request by IhorKoba**
## Purpose

Refactor File Extensions view to use final-form instead of redux-form

## Approach

- Update `FileExtensionForm` component to use stripes-final-form instead of redux-form
- Update dependencies in `package.json`
- Add `composeValidators` helper function to use multiple validation for one field

## Refs

https://issues.folio.org/browse/UIDATIMP-825

### CODE REVIEW

File: src/settings/FileExtensions/FileExtensionForm.js [hunk 1]
```diff
@@ -122,38 +122,46 @@ const FileExtensionFormComponent = ({
...
         <p className={css.checkBoxLabel}>
           <FormattedMessage id="ui-data-import.settings.fileExtension.blockImport" />
         </p>
         <Field
-          label={<FormattedMessage id="ui-data-import.settings.fileExtension.blockImport" />}
           name="importBlocked"
           type="checkbox"
-          component={Checkbox}
-          onChange={handleImportBlockedChange}
+          render={importBlockedProps => (
```
Comment (mariia-aloshyna) [inline src/settings/FileExtensions/FileExtensionForm.js:135]:
```suggestion
          render={importBlockedFieldProps => (
```

File: src/settings/FileExtensions/FileExtensionForm.js [hunk 2]
```diff
@@ -122,38 +122,46 @@ const FileExtensionFormComponent = ({
...
+            <Checkbox
+              {...importBlockedProps}
+              value={form.getState()?.values?.importBlocked}
```
Comment (mariia-aloshyna) [inline src/settings/FileExtensions/FileExtensionForm.js:138]: Isn't `getState()` the required prop for the `form`?

File: src/settings/FileExtensions/FileExtensionForm.js [hunk 3]
```diff
@@ -122,38 +122,46 @@ const FileExtensionFormComponent = ({
...
-          required={dataTypesRequired}
-          disabled={!dataTypesRequired}
+          validate={validateDataTypes}
           validationEnabled
-          itemToString={identity}
-          validate={[validateDataTypes]}
-          filter={filterMultiSelect}
-          formatter={MultiSelectItem}
-          onBlur={handleMultiSelectBlur}
+          render={dataTypesProps => (
```
Comment (mariia-aloshyna) [inline src/settings/FileExtensions/FileExtensionForm.js:150]:
```suggestion
          render={dataTypesFieldProps => (
```

File: src/settings/FileExtensions/FileExtensionForm.js (hunk 1 above)
Comment (IhorKoba) [reply src/settings/FileExtensions/FileExtensionForm.js:135]: done

File: src/settings/FileExtensions/FileExtensionForm.js (hunk 2 above)
Comment (IhorKoba) [reply src/settings/FileExtensions/FileExtensionForm.js:138]: done

File: src/settings/FileExtensions/FileExtensionForm.js (hunk 3 above)
Comment (IhorKoba) [reply src/settings/FileExtensions/FileExtensionForm.js:150]: done

**Review by mariia-aloshyna — APPROVED** [review]

**Review by IvanKryzhanovskyi — APPROVED** [review]

**Pull request merged**